### PR TITLE
Adjustment to meet reporter specification

### PR DESCRIPTION
**Pull request by Qizot**
Minor adjustments to suit reporter specification at https://hexdocs.pm/telemetry_metrics/writing_reporters.html.

- **Reporters MUST honor keep recording rule functions** - has been ignored.

### CODE REVIEW

File: lib/timescaledb/reporter.ex [hunk 1]
```diff
@@ -6,12 +6,17 @@ defmodule Membrane.Telemetry.TimescaleDB.Reporter do
...
 
   @log_prefix "[#{__MODULE__}]"
 
   @spec start_link(any) :: :ignore | {:error, any} | {:ok, pid}
-  def start_link(_) do
-    GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
+  def start_link(opts) do
+    metrics =
+      opts[:metrics] ||
+        raise ArgumentError, "the :metrics options is required by #{inspect(__MODULE__)}"
```
Comment (bblaszkow06) [inline lib/timescaledb/reporter.ex]:
```suggestion
        raise ArgumentError, "the `:metrics` options is required by #{inspect(__MODULE__)}"
```

File: lib/timescaledb/reporter.ex [hunk 2]
```diff
@@ -29,8 +34,9 @@ defmodule Membrane.Telemetry.TimescaleDB.Reporter do
   end
 
   def send_measurement(_) do
-    raise ArgumentError,
-          "#{__MODULE__}: Invalid measurement format, expected map %{element_path: String.t(), method: String.t(), value: integer()"
+    Logger.warn(
+      "#{__MODULE__}: Invalid measurement format, expected map %{element_path: String.t(), method: String.t(), value: integer()}"
```
Comment (bblaszkow06) [inline lib/timescaledb/reporter.ex]:
```suggestion
      "#{__MODULE__}: Invalid measurement format, expected map `%{element_path: String.t(), method: String.t(), value: integer()}`"
```

File: lib/timescaledb/telemetry_handler.ex [hunk 3]
```diff
@@ -18,19 +18,39 @@ defmodule Membrane.Telemetry.TimescaleDB.TelemetryHandler do
...
+      &handle_event/4,
       nil
     )
   end
+
+  def unregister_handler() do
+    :telemetry.detach(get_handler_name())
+  end
+
+  def get_handler_name() do
```
Comment (bblaszkow06) [inline lib/timescaledb/telemetry_handler.ex]: Specs & docs are missing

File: lib/timescaledb/telemetry_handler.ex [hunk 4]
```diff
@@ -18,19 +18,39 @@ defmodule Membrane.Telemetry.TimescaleDB.TelemetryHandler do
     Reporter.send_measurement(measurement)
   end
 
+  def handle_event(
```
Comment (bblaszkow06) [inline lib/timescaledb/telemetry_handler.ex:22]: Specs & docs missing

File: lib/timescaledb/reporter.ex [hunk 5]
```diff
@@ -29,8 +34,9 @@ defmodule Membrane.Telemetry.TimescaleDB.Reporter do
   end
 
   def send_measurement(_) do
```
Comment (bblaszkow06) [inline lib/timescaledb/reporter.ex:36]: Specs & docs missing

**Review by bblaszkow06 — CHANGES_REQUESTED** [review]

File: lib/timescaledb/reporter.ex (hunk 5 above)
Comment (Qizot) [reply lib/timescaledb/reporter.ex:36]: Not missing, just not expanded by github.

**Pull request merged**